### PR TITLE
PERF: run user merging task in a background job.

### PR DESCRIPTION
**Pull request by vinothkannans**
Currently, the admin page is timing out while merging the users with lots of posts.

### CODE REVIEW

File: app/assets/javascripts/admin/addon/models/admin-user.js [hunk 1]
```diff
@@ -515,20 +516,26 @@ const AdminUser = User.extend({
...
-          if (/^\/admin\/users\/list\//.test(location)) {
-            DiscourseURL.redirectTo(location);
-          } else {
-            DiscourseURL.redirectTo(
-              `/admin/users/${data.user.id}/${data.user.username}`
-            );
-          }
+      .then((response) => {
+        if (response.success) {
+          // const messageBus = container.lookup("message-bus:main");
```
Comment (eviltrout) [inline app/assets/javascripts/admin/addon/models/admin-user.js:521]: Looks like an extra line left in by mistake.

File: app/jobs/regular/merge_user.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      current_user_id = args[:current_user_id]
+
+      user = User.find_by(id: args[:user_id])
+      target_user = User.find_by(id: args[:target_user_id])
+      current_user = User.find_by(id: args[:current_user_id])
+      guardian = Guardian.new(current_user)
+      serializer_opts = { root: false, scope: guardian }
+
+      if user = UserMerger.new(user, target_user, current_user).merge!
+        sleep(10)
```
Comment (eviltrout) [inline app/jobs/regular/merge_user.rb:17]: I assume this line was also used for testing?

File: app/services/user_merger.rb [hunk 3]
```diff
@@ -356,6 +378,8 @@ def update_user_ids
   end
 
   def delete_source_user
+    ::MessageBus.publish '/merge_user', { message: I18n.t("admin.user.merge_user.deleting_source_user") }
```
Comment (eviltrout) [inline app/services/user_merger.rb:381]: I like how it updates the user as it goes along!

File: app/services/user_merger.rb [hunk 4]
```diff
@@ -30,6 +31,9 @@ def merge!
   protected
 
   def update_username
+    return if @source_user.username == @target_user.username
+
+    ::MessageBus.publish '/merge_user', { message: I18n.t("admin.user.merge_user.updating_username") }
```
Comment (gschlager) [inline app/services/user_merger.rb:36]: I think the `MessageBus.publish` calls should be scoped to the `acting_user` or is there a reason for those messages to be available for all users?

Comment (eviltrout) [reply app/services/user_merger.rb:36]: Oh very good catch @gschlager ! Yes this must be fixed.

File: app/jobs/regular/merge_user.rb (hunk 2 above)
Comment (vinothkannans) [reply app/jobs/regular/merge_user.rb:17]: oops! yes.

**Review by lis2 — APPROVED** [review]

**Pull request merged**